diagram.md

```diff
@@ -191,28 +193,39 @@ The `Cluster Elements` contain all of the video frames and audio for each track.
+|       |            |-------------|
+|       |            | Timecode    |
+|       |            |------------------------+
+|       |            | Flags       | Gap      |
+|       |            |             |----------|
+|       |            |             | Lacing   |
+|       |            |             |----------|  
+|       |            |             | Reserved |
+|       |-------------------------------------+
+|       | Optional   | FrameSize   |
+|       |--------------------------|
+|       | Data       | Frame       |  
++----------------------------------+
+```
+Figure: Representation of the `Block Element` structure.  
+
+Each `Cluster` MUST contain a `Timecode Element`. The `Timecode Element` value MUST be stored once per `Cluster`. The `Timecode Element` in the `Cluster` is relative to the entire `Segment`. The `Timecode Element` SHOULD be the first `Element` in the `Cluster`.
+
+Additionally, the `Block` contains an offset that, when added to the `Cluster`'s `Timecode Element` value, yields the `Block`'s effective timecode. Therefore, timecode in the `Block` itself is relative to the `Timecode Element` in the `Cluster`. For example, if the `Timecode Element` in the `Cluster` is set to 10 seconds and a `Block` in that `Cluster` is supposed to be played 12 seconds into the clip, the timecode in the `Block` would be set to 2 seconds.
 
-* Portion of Block
-  * Data Type
-     * Bit Flag
-* Header
-  * TrackNumber
-  * Timecode
-  * Flags
-    * Gap
-    * Lacing
-    * Reserved
-* Optional
-  * FrameSize
-* Data
-  * Frame
+The `ReferenceBlock` in the `BlockGroup` is used instead of the basic "P-frame"/"B-frame" description. Instead of simply saying that this `Block` depends on the `Block` directly before, or directly afterwards, the `Timecode` of the necessary `Block` is used. Because there can be as many `ReferenceBlock Elements` as necessary for a `Block`, it allows for some extremely complex referencing.
 
-Although the Timecode value is stored once per Cluster, another timecode is stored within the Block structure itself. The way this works is that the Timecode in the Cluster is relative to the entire `Segment`. It is usually the Timecode that the first Block in the Cluster needs to be played at. The Timecode in the Block itself is relative to the Timecode in the Cluster. For example, let's say that the Timecode in the Cluster is set to 10 seconds, and you have a Block in that Cluster that is supposed to be played 12 seconds into the clip; this means that the Timecode in the Block would be set to 2 seconds.
+The `Cues Element` is used to seek when playing back a file by providing a temporal index for each of the `Tracks`. It is similar to the `SeekHead Element`, but used for seeking to a specific time when playing back the file. It is possible to seek without this element, but it is much more difficult because the player has to 'hunt and peck' through the file looking for the correct timecode.
```

Comment:
It's not "each of the Tracks" but only "some of the Tracks", usually on the video track(s).